app/assets/javascripts/accounts.js

```diff
@@ -1,40 +1,32 @@
 $(function(){
-  function toggleInputs() {
-    var class_select = $("#class_type");
-    var class_type   = class_select.val();
-
-    $('.account_type_fields').each(function() {
-      var visible = $(this).hasClass(class_type);
-      $(this).toggle(visible);
-      $(this).find('input, select, textarea').prop('disabled', !visible);
-    });
-
-    $('.' + class_type + ' .affiliate').trigger('change');
-  }
 
+  // ---------------------------------------------------------------------------
+  // Toggle affiliate
+  // ---------------------------------------------------------------------------
   function toggleAffiliate() {
-    $('.affiliate_other').toggle($('.affiliate:visible option:selected').text() == 'Other')
+    $(".js--affiliate_other").toggle($(".js--affiliate:visible option:selected").text() == "Other")
   }
 
-  $(".datepicker").datepicker({minDate:+0, maxDate:'+3y', dateFormat: 'mm/dd/yy'});
-
-  $("#class_type").change(function() {
-    toggleInputs();
-  });
-
-  $('.affiliate').change(function() {
+  $(".js--affiliate").change(function() {
     toggleAffiliate();
   });
 
-  toggleInputs();
+  toggleAffiliate();
+
+  // ---------------------------------------------------------------------------
+  // Date picker
+  // ---------------------------------------------------------------------------
+  $(".datepicker").datepicker({minDate:+0, maxDate:"+3y", dateFormat: "mm/dd/yy"});
 
+  // ---------------------------------------------------------------------------
   // Autotab
-  $('.account_number_field :input[maxlength]').keyup(function(evt) {
+  // ---------------------------------------------------------------------------
+  $(".account_number_field :input[maxlength]").keyup(function(evt) {
```

Comment:
I know it's not new, but what does `:input` do? Does that also include `select`? Does that matter here because of `maxlength`.